build.gradle

```diff
@@ -80,7 +80,7 @@ subprojects {
     compileClasspath {
       // do not exclude Guava so the bundle project can reference classes.
       // the Spark module is also excluded because this breaks the Scala compiler
-      if (project.name != 'iceberg-bundled-guava' && project.name != 'iceberg-spark') {
+      if (project.name != 'iceberg-bundled-guava' && project.name != 'iceberg-spark' && project.name != 'iceberg-spark2') {
```

Comment:
we do not exclude guava for spark module?

Reply:
Also should this be iceberg-spark3 instead of iceberg-spark?

Reply:
Not having Guava in the compile classpath breaks Scala compilation. Right now, I'm planning to avoid adding Scala code to Spark 3, which is why it is only for the current module and for the Spark 2 module where I plan to move the current Scala code. See #1088.